add-on/src/recovery/recovery.js

```diff
@@ -0,0 +1,66 @@
+'use strict'
+/* eslint-env browser, webextensions */
+
+import choo from 'choo'
+import html from 'choo/html/index.js'
+import { i18n, runtime } from 'webextension-polyfill'
+import { renderLogo } from '../landing-pages/welcome/page.js'
+import createWelcomePageStore from '../landing-pages/welcome/store.js'
+import './recovery.css'
+
+const app = choo()
+
+// TODO (whizzzkid): refactor base store to be more generic.
+app.use(createWelcomePageStore(i18n, runtime))
+// Register our single route
+app.route('*', (state) => {
+  const { hash } = window.location
+  const { href: publicURI } = new URL(decodeURIComponent(hash.slice(1)))
+  if (!publicURI) {
+    return
+  }
+
+  const openURLFromHash = () => {
+    try {
+      console.log('Opening URL from hash:', publicURI)
+      window.location.href = publicURI
```

Comment:
[`window.location.replace`](https://developer.mozilla.org/en-US/docs/Web/API/Location/replace) is better for things like this, because it will replace the URL in browser history, so the back button will no longer point at an interstitial page.

```suggestion
      window.location.replace(publicURI)
```